### PR TITLE
agent: Upgrade tokio to 1.2.0

### PR DESCRIPTION
**Pull request by Tim-Zhang**
Fixes: #1257

Signed-off-by: Tim Zhang <tim@hyper.sh>

### CODE REVIEW

**Comment by fidencio** [comment]
I'm really sorry for the disruption but in order to do the branch name changes announced [here](http://lists.katacontainers.io/pipermail/kata-dev/2021-January/001695.html) I'm adding the "do-not-merge" and the "needs-branch-adjustment" labels to the PR.

I'll add a new comment once the branch changes are done and once it happens I'd like to ask you to resubmit the PR against the new `main` branch, possibly linking to this PR in case you find the conversations / reviews already made important to proceed.

**Comment by fidencio** [comment]
Kata Containers repos are back to business.  Thanks a lot for your patience and I'm very happy to announce that there's no changes needed from your side as renaming the branches automagically reflected here.

If you notice something weird, some missing CI, please, do not hesitate on contacting us here or on #kata-dev on bit.ly/katacontainersslack.

Also, this is a good opportunity to have your PR rebased atop of `main`, to ensure your PR follow all the guidelines (https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format) enforced by this community, or even to re-evaluate whether the PR is currently needed.

Last but not least,  as the default branch has been renamed to `main`, if you have a local clone you'd need to update it and you do so by running:
```
git branch -m 2.0-dev main
git fetch origin
git branch -u origin/main main
```

Happy hacking!

File: src/agent/Cargo.toml [hunk 1]
```diff
@@ -22,11 +22,11 @@ scopeguard = "1.0.0"
 regex = "1"
 
 async-trait = "0.1.42"
-tokio = { version = "0.2", features = ["rt-core", "sync", "uds", "stream", "macros", "io-util", "time", "signal", "io-std", "process", "blocking"] }
-futures = "0.3"
-netlink-sys = { version = "0.4.0", features = ["tokio_socket",]}
-tokio-vsock = "0.2.2"
-rtnetlink = "0.6.0"
+tokio = { version = "1.0.1", features = ["rt", "sync", "macros", "io-util", "time", "signal", "io-std", "process"] }
```
Comment (jodh-intel) [inline src/agent/Cargo.toml:25]: Tokio is at 1.1.0 now.

**Review by jodh-intel — COMMENTED** [review]
Hi @Tim-Zhang - could you updated for Tokio 1.1.0 (now)?

File: src/agent/Cargo.toml (hunk 1 above)
Comment (Tim-Zhang) [reply src/agent/Cargo.toml:25]: OK, will do that before ready to review.

Comment (jodh-intel) [reply src/agent/Cargo.toml:25]: Any update on this @Tim-Zhang? The shutdown + tracing work (#411) needs the latest tokio code so this is a blocker for tracing.

/cc @liubin.

File: src/agent/src/netlink.rs [hunk 2]
```diff
@@ -332,6 +332,16 @@ impl Handle {
...
+                .route()
+                .add()
+                .table(MAIN_TABLE)
+                .kind(UNICAST)
+                .protocol(BOOT_PROT)
+                .scope(scope);
+
             // `rtnetlink` offers a separate request builders for different IP versions (IP v4 and v6).
             // This if branch is a bit clumsy because it does almost the same.
             // TODO: Simplify this once https://github.com/little-dude/netlink/pull/140 is merged and released
```
[the remark this answers is not in the document]
Comment (Tim-Zhang) [reply src/agent/src/netlink.rs]: The TODO has been removed, thx.

File: src/agent/Cargo.toml (hunk 1 above)
Comment (Tim-Zhang) [reply src/agent/Cargo.toml:24]: @jodh-intel The pr is ready to review now, PTAL, Thanks

**Review by liubin — APPROVED** [review]

File: src/agent/Cargo.toml (hunk 1 above)
Comment (jodh-intel) [reply src/agent/Cargo.toml]: Ping @Tim-Zhang, or @lifupan, @liubin :smile:

**Review by mxpv — APPROVED** [review]

**Review by lifupan — APPROVED** [review]

**Pull request merged**